enhancements/installer/credentials-management-outside-openshift-cluster.md

```diff
@@ -0,0 +1,130 @@
+creation-date: 2021-03-25
+last-updated: 2021-03-25
+status: provisional
+---
+
+# Credentials Management outside OpenShift Cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to take the process of credentials management outside the OpenShift cluster for new 
+platforms. We propose to make *manual* mode as default for clusters on new platforms. 
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy the customer requirement to follow the best security practices 
+by not storing admin credentials inside the cluster
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Set *manual* as default/preferred credentials mode for OpenShift
+* Not store admin level cloud credentials inside a cluster
+* Provide a tooling design for credentials management outside the cluster
+
+### Non-Goals
+
+* We do not plan to go into design details on how this tool will be ported to cloud providers other than AWS.
+
+## Proposal
+
+Currently, *mint* is a default credentials mode for OpenShift. In this mode we run OpenShift installer with an admin 
+level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential 
+operator to process the CredentialRequests in the cluster and create new users with fine-grained permissions.
+The customers have reported that the need to store admin credentials inside the cluster is a major disadvantage. Based 
+on the feedback, we propose to run credentials related setup outside a cluster and then start installation process in a 
+*manual* mode without providing admin credentials to the installer.
+
+### User Stories
+
+#### Story 1
+As a OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image
+and create credentials in the underlying cloud provider. 
+
+#### Story 2
+As a OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image 
+and create Kubernetes Secrets in to satisfy all in-cluster component's CredentialsRequests.
+
+#### Story 3
+As a OpenShift cluster administrator I should be able to install a OpenShift cluster, without providing admin credentials
+to the installer, by injecting precreated Kubernetes Secrets.
+
+### Risks and Mitigations
+
+## Design Details
+
+We intend to build an optional tool **ccoctl** which will handle credentials management of the cluster in *manual* mode. 
+The following is the set of requirements for the current prototype (for AWS). We can have something similar for other 
+platforms.
```

Comment:
I think we need a plan here for how ccoctl is distributed. I am really wondering if this should be exposed in oc adm.

Reply:
i assume distribution of `ccoctl` is specific to an enhancement around that component.  this enhancement is more focused on 'should each cluster be empowered to mint scoped cloud credentials' as a default posture moving forward.